packages/contracts/deploy/40_finalize-managing-dao/10_install-multisig-on-managing-dao.ts

```diff
@@ -0,0 +1,137 @@
+import {HardhatRuntimeEnvironment} from 'hardhat/types';
+import {DeployFunction} from 'hardhat-deploy/types';
+
+import buildMetadataJson from '../../src/plugins/governance/multisig/build-metadata.json';
+import {findEvent} from '../../utils/event';
+import {hashHelpers} from '../../test/test-utils/psp/hash-helpers';
+
+import {getContractAddress} from '../helpers';
+
+interface Ehre extends HardhatRuntimeEnvironment {
+  aragonPluginRepos: {multisig: string};
+}
+
+const func: DeployFunction = async function (hre: HardhatRuntimeEnvironment) {
+  const {getNamedAccounts, ethers} = hre;
+  const {deployer} = await getNamedAccounts();
+
+  const ehre = hre as Ehre;
+
+  // Get info from .env
+  const approvers =
+    process.env.MANAGINGDAO_MULTISIG_APPROVERS?.split(',') || [];
+  const minApprovals = process.env.MANAGINGDAO_MULTISIG_MINAPPROVALS || 0;
+  const listedOnly = process.env.MANAGINGDAO_MULTISIG_LISTEDONLY || false;
+
+  if (!approvers.length || !minApprovals || !listedOnly)
+    throw new Error(
+      `Some .env settings for managingDAO multisig are not set correctly, see:\n` +
+        `(MANAGINGDAO_MULTISIG_APPROVERS no. of addresses: ${approvers.length})\n` +
+        `(MANAGINGDAO_MULTISIG_MINAPPROVALS: ${minApprovals})\n` +
+        `(MANAGINGDAO_MULTISIG_LISTEDONLY: ${listedOnly})\n`
+    );
+
+  // Get `managingDAO` address.
+  const managingDAOAddress = await getContractAddress('DAO', hre);
```

Comment:
Are we 100% sure that this will give us the managing DAO address? If it returns the last address recorded, there might have been another step deploying a DAO and we don't automatically know

Reply:
yes, 
hardhat will reserve the name DAO and assign the address to it,
if someone during deploy tries to redeploy dao, it will reuse the address and it will not deploy it again